packages/filesystem/src/browser/file-tree/file-tree-widget.ts

```diff
@@ -56,4 +63,108 @@ export class FileTreeWidget extends TreeWidget {
         return super.decorateCaption(node, VirtualRenderer.merge(icon, caption), props);
     }
 
+    protected onAfterAttach(msg: Message): void {
+        super.onAfterAttach(msg);
+        if (this.model.root !== undefined) {
+            const treeNode = this.model.root;
+            this.addEventListener(this.node, 'dragenter', event => this.handleDragEnterEvent(treeNode, event));
+            this.addEventListener(this.node, 'dragover', event => this.handleDragOverEvent(treeNode, event));
+            this.addEventListener(this.node, 'dragleave', event => this.handleLeaveEvent(event));
+            this.addEventListener(this.node, 'drop', event => this.handleDropEvent(treeNode, event, false));
+        }
+    }
+
+    protected createNodeAttributes(node: ITreeNode, props: NodeProps): ElementAttrs {
+        const elementAttrs: ElementAttrs = super.createNodeAttributes(node, props);
+        const childExpansion = this.isExpandable(node) ? 0 : 1;
+
+        if (!childExpansion) {
```

Comment:
I will be better to check with `DirNode.is(node)`, since expandable not necessary means that it is not a file

Reply:
OK , same as above, using DirNode.is()